modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaFetchReferenceCodeMethodCheck.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Micaelle Silva
+ */
+public class UpgradeJavaFetchReferenceCodeMethodCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			String newJavaMethodContent = javaMethodContent;
+
+			Matcher matcher = _fetchExternalReferenceCodePattern.matcher(
+				javaMethodContent);
+
+			while (matcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, matcher.start());
+
+				String variableName = getVariableName(methodCall);
+
+				variableName = variableName.trim();
+
+				if (!variableName.equals("CPInstanceLocalServiceUtil") &&
+					!variableName.equals("CPInstanceServiceUtil") &&
+					!hasClassOrVariableName(
+						"CPInstanceLocalService", content, content,
```

Comment:
@MicaelleSilvaa, we should use the `javaMethodContent` instead of `content` after the name of the service on the `hasClassOrVariableName` parameters. The first `content` that we pass as a `parameter` is the `content` of the actual context (we are dealing with methods, so should be the content of the method. The second one is the file's complete `content`).

Reply:
We should do that on both `hasClassOrVariableName` calls